recorder_test.go

```diff
@@ -3,185 +3,277 @@ package lightstep
-	for _, span := range spans {
-		recorder.RecordSpan(span)
-	}
+			Context("When an object is translated", func() {
+				It("Is marshled into JSON", func() {
+					for i := 0; i < 8; i++ {
+						pl, _ := json.Marshal([]interface{}{i, i, true, "suhhh"})
+						Expect(logs[i].Keyvalues[1]).To(Equal(
+							&cpb.KeyValue{Key: "object", Value: &cpb.KeyValue_JsonValue{string(pl)}},
+						))
+					}
+				})
+			})
 
-	checkCapSize(maxBuffer, maxBuffer)
+			Context("When a long string field is translated", func() {
+				It("Is translated to a truncated String Key Value pair", func() {
+					for i := 0; i < 8; i++ {
+						Expect(logs[i].Keyvalues[2]).To(Equal(
+							&cpb.KeyValue{
+								Key:   "too_long-----------…",
+								Value: &cpb.KeyValue_StringValue{"---------------------------------------…"},
+							},
+						))
+					}
+				})
+			})
+		})
+	})
 
-	_ = NewTracer(Options{
-		AccessToken: "0987654321",
-		UseThrift:   true,
-	}).(basictracer.Tracer).Options().Recorder.(*thrift_rpc.Recorder)
-}
+	Context("When calling Close() twice", func() {
+		Context("With grpc enabled", func() {
+			It("Should not fail", func() {
+				tracer = NewTracer(Options{
+					AccessToken: "0987654321",
+					UseGRPC:     true,
+				})
+				err := CloseTracer(tracer)
+				Expect(err).NotTo(HaveOccurred())
+
+				err = CloseTracer(tracer)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
 
-func TestDoubleClose(t *testing.T) {
-	tracer := NewTracer(Options{
-		AccessToken: "0987654321",
-		UseGRPC:     true,
+		Context("With thrift enabled", func() {
+			It("Should not fail", func() {
+				tracer = NewTracer(Options{
+					AccessToken: "0987654321",
+					UseGRPC:     false,
+				})
+				err := CloseTracer(tracer)
+				Expect(err).NotTo(HaveOccurred())
+
+				err = CloseTracer(tracer)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
 	})
-	err := CloseTracer(tracer)
-	if err != nil {
-		t.Errorf("err on first close: %s", err.Error())
-	}
-	err = CloseTracer(tracer)
-	if err != nil {
-		t.Errorf("err on second close: %s", err.Error())
-	}
-}
 
-func TestDoubleCloseThrift(t *testing.T) {
-	tracer := NewTracer(Options{
-		AccessToken: "0987654321",
-		UseGRPC:     false,
+	Describe("Regarding the span buffer", func() {
+		Context("With the default max span buffer size", func() {
+			BeforeEach(func() {
+				recorder = NewTracer(Options{
+					AccessToken: "0987654321",
+					UseGRPC:     true,
+				}).(basictracer.Tracer).Options().Recorder.(*Recorder)
+			})
+
+			Context("Before any spans have been started", func() {
+				It("The buffer should be empty and the capacity should be the default", func() {
+					recorder.lock.Lock()
+					defer recorder.lock.Unlock()
+					Expect(cap(recorder.buffer.rawSpans)).To(Equal(defaultMaxSpans))
+					Expect(len(recorder.buffer.rawSpans)).To(Equal(0))
+				})
+			})
+
+			Context("After spans have been started", func() {
+				It("The buffer should respond appropriately", func() {
```

Comment:
This large It block actually covers two contexts: "When the buffer is at capacity" and "When the buffer is over capacity". Please extract these contexts, and extract appropriate `BeforeEach` blocks that set up the context.